rct229/rulesets/ashrae9012019/section19/section19rule20.py

```diff
@@ -0,0 +1,199 @@
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        rmi_p = context.proposed
+
+        return {
+            "zone_supply_return_exhaust_relief_terminal_fan_power_dict_p": get_zone_supply_return_exhaust_relief_terminal_fan_power_dict_p(
+                rmi_p
+            ),
+            "dict_of_zones_and_terminal_units_served_by_hvac_sys_b": get_dict_of_zones_and_terminal_units_served_by_hvac_sys(
+                rmi_b
+            ),
+        }
+
+    class HVACRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section19Rule20.HVACRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={
+                    "$": ["fan_system"],
+                },
+            )
+
+        def get_calc_vals(self, context, data=None):
+            hvac_b = context.baseline
+            hvac_id_b = hvac_b["id"]
+            zone_supply_return_exhaust_relief_terminal_fan_power_dict_p = data[
+                "zone_supply_return_exhaust_relief_terminal_fan_power_dict_p"
+            ]
+            list_zones_served_b = data[
+                "dict_of_zones_and_terminal_units_served_by_hvac_sys_b"
+            ][hvac_id_b]["zone_list"]
+
+            fan_sys_b = hvac_b["fan_system"]
+
+            fan_sys_power_flow_info_dict_b = (
+                get_fan_system_object_supply_return_exhaust_relief_total_power_flow(
+                    fan_sys_b
+                )
+            )
+
+            hvac_sys_total_supply_fan_power_b = fan_sys_power_flow_info_dict_b[
+                "supply_fans_power"
+            ]
+            hvac_sys_total_return_fan_power_b = fan_sys_power_flow_info_dict_b[
+                "return_fans_power"
+            ]
+            hvac_sys_total_exhaust_fan_power_b = fan_sys_power_flow_info_dict_b[
+                "exhaust_fans_power"
+            ]
+            hvac_sys_total_relief_fan_power_b = fan_sys_power_flow_info_dict_b[
+                "relief_fans_power"
+            ]
+
+            total_modeled_fan_power_b = (
+                hvac_sys_total_supply_fan_power_b
+                + hvac_sys_total_return_fan_power_b
+                + hvac_sys_total_exhaust_fan_power_b
+                + hvac_sys_total_relief_fan_power_b
+            )
+
+            proposed_total_supply_fan_power = ZERO.POWER
+            proposed_total_return_fan_power = ZERO.POWER
+            proposed_total_exhaust_fan_power = ZERO.POWER
+            proposed_total_relief_fan_power = ZERO.POWER
+            for zone_id_b in list_zones_served_b:
+                proposed_total_supply_fan_power += (
+                    zone_supply_return_exhaust_relief_terminal_fan_power_dict_p[
+                        zone_id_b
+                    ]["zone_supply_fan_power"]
```

Comment:
The function return value is supply_fan_power

zone_supply_return_exhaust_relief_terminal_fan_power_dict[zone["id"]] = {
            "supply_fans_power": zone_total_supply_fan_power,
            "return_fans_power": zone_total_return_fan_power,
            "exhaust_fans_power": zone_total_exhaust_fan_power,
            "relief_fans_power": zone_total_relief_fan_power,
            "terminal_fans_power": zone_total_terminal_fan_power,
        }

Reply:
Addressed.